### ISSUE TITLE
Enabling systemd on WSL Doesn't Work as Expected

### ISSUE THREAD

**Issue by arinanto** [AlmaLinux/wsl-images#30]
# Brief description

Recently Microsoft enabled `systemd` support on WSL ([more info on MS Devblogs](https://devblogs.microsoft.com/commandline/systemd-support-is-now-available-in-wsl/)). Trying to enable `systemd` on AlmaLinux 9 inside WSL result in partial activation where `systemctl --user` not working.

# How to replicate

 - Make sure WSL is running version 0.67.6 or higher (`wsl --version`)
 - Configure `/etc/wsl.conf` to enable `systemd` support
   ```
   [boot]
   systemd=true
   ```
 - Run `systemctl --user status` inside WSL

# Test result

## WSL version (`wsl --version`)

```
WSL version: 0.70.0.0
Kernel version: 5.15.68.1
WSLg version: 1.0.45
MSRDC version: 1.2.3575
Direct3D version: 1.606.4
DXCore version: 10.0.25131.1002-220531-1700.rs-onecore-base2-hyp
Windows version: 10.0.22621.674
```

## Actual behaviour on AlmaLinux 9

![AlmaLinux 9 screenshot](https://user-images.githubusercontent.com/64170577/196349716-2e1c1c07-d280-41b3-9835-e85400041b88.png)

## Behaviour observed on Ubuntu 22.04

![Ubuntu 22.04 screenshot](https://user-images.githubusercontent.com/64170577/196349927-73881b49-e754-4446-ac35-e05787907690.png)

**Comment by tanero** [comment]
same issue, is there any update ?

**Comment by LKHN** [comment]
> same issue, is there any update ?

Hi Taner,

We recently released a new version of AlmaLinux OS 9 as `9.4.1.0` and AlmaLinux OS 8 as `8.10.1.0`.

Starting with the new images the systemd is enabled by default and fully functional without any issues.
Here is full list of changes on these versions:

AlmaLinux OS 9 (`9.4.1.0`):

>- Added Windows Terminal integration with icon and color scheme support.
>- Added systemd support.
>- Switched from passwordless sudo to sudo with providing current user password by default.
>- Distribution Name is changed to AlmaLinux-9.

AlmaLinux OS 8 (`8.10.1.0`):

>- Added Windows Terminal integration with icon and color scheme support.
>- Added systemd support.
>- Switched from passwordless sudo to sudo with providing current user password by default.


Since the news images introduce some breaking changes like the change on tge namings.
Which means You need to:
1. Take backup of your files and configurations from the current AlmaLinux OS 9/8 distro.
2. (Unregister or uninstall a Linux distribution)[https://learn.microsoft.com/en-us/windows/wsl/basic-commands#unregister-or-uninstall-a-linux-distribution]
3. Uninstall AlmaLinux OS 9/AlmaLinux OS 8 WSL apps from your system.

Then you can install the new AlmaLinux OS 9 and AlmaLinux OS 8 images, enjoy the systemd out-of-the box and fully integration Windows Terminal support.

See the screenshots how the Windows Terminal integrartion works now at:
- AlmaLinux OS 9: https://apps.microsoft.com/detail/9p5rwlm70sn9?ocid=webpdpshare
- AlmaLinux OS 8: https://apps.microsoft.com/detail/9p5rwlm70sn9?ocid=webpdpshare


We also working with WSL team at Microsoft to add support of [installation](https://learn.microsoft.com/en-us/windows/wsl/basic-commands#install) of AlmaLinux OS 9 and AlmaLinux OS 8 via `wsl` command line, such as:

AlmaLinux OS 9:
```powershell
wsl --install AlmaLinux-9
```

AlmaLinux OS 8:
```powershell
wsl --install AlmaLinux-8 
```

This is an alternative method of installation our wsl images where the Microsoft App Store is not available (Windows Server 2019) or restricted (Windows 10 and 11).

Once the wsl command integration is done, We will publish a blog post to cover all changes on the new images.

Please let us know how you feel about the images here or on  https://chat.almalinux.org/almalinux/channels/sigwsl :)

**Comment by tanero** [comment]
thanks for detailed response, I have tried many things, but no luck, systemd is still not working.

```shell
wsl --version
WSL version: 2.2.4.0
Kernel version: 5.15.153.1-2
WSLg version: 1.0.61
MSRDC version: 1.2.5326
Direct3D version: 1.611.1-81528511
DXCore version: 10.0.26091.1-240325-1447.ge-release
Windows version: 10.0.19045.4894
``` 

and /etc/wsl.conf

```shell
[boot]
systemd=true
``` 

```shell
 sudo ps aux
USER       PID %CPU %MEM    VSZ   RSS TTY      STAT START   TIME COMMAND
root         1  0.0  0.0  10532   712 ?        Ssl  18:14   0:00 /init
root         5  0.0  0.0  10516   588 ?        Sl   18:14   0:00 plan9 --control-socket 6 --log-level 4 --server-fd 7 --pipe-fd 9 --socket-path
root         8  0.0  0.0  10560   360 tty1     Ss   18:14   0:00 /init
```

**Comment by LKHN** [comment]
Taner,

Strange. Is it the latest image from the Microsoft App Store?

**Comment by tanero** [comment]
> Taner,
> 
> Strange. Is it the latest image from the Microsoft App Store?

yes, wsl and linux image are latest version.

**Comment by LKHN** [comment]
What is the output of

```sh
wsl -l -v
```

**Comment by tanero** [comment]
```shell
wsl -l -v
  NAME           STATE           VERSION
* AlmaLinux-9    Stopped         1
```

**Comment by LKHN** [comment]
Taner,

Oh you are using WSL1 not the WSL2.

Please, use the instructions here to enable WSL2 for `AlmaLinux-9` or as a default WSL version for the all WSL distros.

https://learn.microsoft.com/en-us/windows/wsl/basic-commands#set-wsl-version-to-1-or-2

**Comment by LKHN** [comment]
Taner,

Additionally, the systemd support for WSL2-only is not an AlmaLinux OS specific. See the comparison here between the WSL1 vs WSL2.

As the official WSL documentation mentions, you can face issue while conversion a distro between the WSL versions. Setting default. Completely removing the distro, setting default WSL version and then clean installing again should solve any issues.

https://learn.microsoft.com/en-us/windows/wsl/compare-versions

**Comment by tanero** [comment]
> As the official WSL documentation mentions, you can face issues while conversion a distro between the WSL versions.
> 
> 1. Completely removing the distro (Unregister + Uninstall).
> 2. Setting default WSL version as WSL2
> 3. Installing the distro again should solve any possible issues.

thank you so much, it finally worked.

**Comment by LKHN** [comment]
@tanero,

Thanks a lot for the verification that systemd is enabled by default and works. Please, do not hesitate if you face any issues or see any point for improvements. Enjoy!

@arinanto,

I think we can close this issue as completed. What do you think?

**Comment by arinanto** [comment]
I'm currently not using this WSL distro thus can't try it for myself. That said, @tanero already got it working so it should work. I close this for now. Thank you for resolving this issue.

**Comment by arm-moodys** [comment]
instead of creating a new issue I wanted to share that I ran into the same issue.  I have a fresh Windows 11 with WSL version 2 and I downloaded the latest AlmaLinux9:

1. I added the conf below in the `/etc/wsl.conf` file:
```
[boot]
systemd=true
```
1. shutdown the instance and waited 15 seconds (WSL recommend 8 seconds)
1. I noticed that Alma is now labeled as`AlmaLinux-9` and not `AlmaLinuxOS-9` (I have another windows machine where is labeled as this and the problems is not present) as before.
1. If I run a systemctl command I get the following error:

```bash
System has not been booted with systemd as init system (PID 1). Can't operate.
Failed to connect to bus: Host is down
```

could anyone take a look at this?

**Comment by LKHN** [comment]
@arm-moodys ,

Thanks for the report. Is it the latest WSL image? (`9.4.1.0`)
I ask this because the `/etc/wsl.conf` should be exist with the `systemd=true` in it.

**Comment by arm-moodys** [comment]
Thanks for replying back. Everything was configured properly, I always have some automation that sets everything the way I want after installing any WSL distribution I need, the problem this time was that I couldn't access my repo (which I usually have cloned in my windows machine and then run the scripts form the WSL instance and everything runs smoothly) and hence just manually added the conf in the some files in the windows machine and the then ran the scripts from WSL and well, this was the problem as the files I had added manually had a bunch `M^` characters in each line as they were not LF formatted. So, all is good, it was my mistake.
